### PR TITLE
CV2-5005: Sentry issue related to ES

### PR DESCRIPTION
**Pull request by melsawy**
## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context – why has this been changed/fixed.

References: CV2-5005

## How has this been tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can verify the changes. Please describe whether or not you implemented automated tests.

## Things to pay attention to during code review

Please describe parts of the change that require extra attention during code review, for example:

- File FFFF, line LL: This refactoring does this and this. Is it consistent with how it’s implemented elsewhere?
- Etc.

## Checklist

- [x] I have performed a self-review of my own code
- [x] I have added unit and feature tests, if the PR implements a new feature or otherwise would benefit from additional testing
- [ ] I have added regression tests, if the PR fixes a bug
- [ ] I have added logging, exception reporting, and custom tracing with any additional information required for debugging
- [ ] I considered secure coding practices when writing this code. Any security concerns are noted above.
- [ ] I have commented my code in hard-to-understand areas, if any
- [ ] I have made needed changes to the README
- [ ] My changes generate no new warnings
- [ ] If I added a third party module, I included a rationale for doing so and followed our current [guidelines](https://meedan.atlassian.net/wiki/spaces/ENG/overview#Choose-the-%E2%80%9Cright%E2%80%9D-3rd-party-module)

### CODE REVIEW

File: app/models/concerns/project_association.rb [hunk 1]
```diff
@@ -86,26 +86,21 @@ def add_elasticsearch_data
...
+        data[fname] = { method: fname, klass: 'ProjectMedia', id: self.id, type: 'int' } if self.send("saved_change_to_#{fname}?")
+      end
+      data['channel'] = self.channel.values.flatten.map(&:to_i) if self.send(:saved_change_to_channel?)
+      data['source_name'] =  self.source&.name if self.send(:saved_change_to_source_id?)
+      unless data.blank?
+        data['updated_at'] = self.updated_at.utc
+        options = { keys: data.keys, data: data, pm_id: self.id }
+        model = { klass: self.class.name, id: self.id }
+        ElasticSearchWorker.perform_in(1.second, YAML::dump(model), YAML::dump(options), 'update_doc')
+      end
```
Comment (caiosba) [inline app/models/concerns/project_association.rb:103]: @melsawy , it's missing the `title` and `media_published_at` fields in this refactoring.

File: test/models/project_media_3_test.rb [hunk 2]
```diff
@@ -72,6 +72,8 @@ def setup
     pm.refresh_media = true
     pm.save!
     pm = ProjectMedia.find(pm.id)
+    # TODO: discuss the failure with Caio
+    assert_equal 601720260, pm.media_published_at
```
Comment (caiosba) [inline test/models/project_media_3_test.rb:76]: What we need to discuss @melsawy ? :D

**Review by caiosba — CHANGES_REQUESTED** [review]
@melsawy , as far as I understand, your approach here is to save only the fields that actually have changed in that update by that specific request/thread, right?

**Comment by melsawy** [comment]
> @melsawy , as far as I understand, your approach here is to save only the fields that actually have changed in that update by that specific request/thread, right?

Right @caiosba

File: test/models/project_media_3_test.rb (hunk 2 above)
Comment (melsawy) [reply test/models/project_media_3_test.rb]: Fixed the test, thanks Caio

File: app/models/concerns/project_association.rb (hunk 1 above)
Comment (melsawy) [reply app/models/concerns/project_association.rb:103]: `title` and `media_published_at` are cached fields and already we update the ES during cached field updates so no need to include these fields

Comment (caiosba) [reply app/models/concerns/project_association.rb:103]: Ah OK, I thought you just missed them.

**Review by caiosba — APPROVED** [review]

**Pull request merged**